### PR TITLE
Cleanup FindPCRE

### PR DESCRIPTION
**Pull request by Bjoe**
Please test if it builds for you.
Missing part is: Install FindPCRE. I will add this also if this build is fine.

### CODE REVIEW

File: cmake/FindPCRE.cmake [hunk 1]
```diff
@@ -5,29 +5,58 @@
 #  PCRE_INCLUDE_DIRS - where to find pcre.h, etc.
 #  PCRE_LIBRARIES    - List of libraries when using pcre.
 #  PCRE_FOUND        - True if pcre found.
```
Comment (lmayencourt) [inline cmake/FindPCRE.cmake]: just add a comment specifying the new exported target:
`#  Pcre::Pcre        - Imported target for pcre, if pcre has been found.`

Comment (lmayencourt) [reply cmake/FindPCRE.cmake]: Build fine for me.
The target_properties of the Poco::Foundation target seems right too: `INTERFACE_LINK_LIBRARIES "Pcre::Pcre;ZLIB::ZLIB;pthread;dl;rt" `

**Comment by lmayencourt** [comment]
Thanks for the update ! I have tested the installation and it seems ok !
I still want to try the cross-compilation use-case against the installed files (building ROS2 in my case), which was the reason for this changes.

**Comment by lmayencourt** [comment]
Thanks for the update ! 
I have tested by cross-compiling an application (building ROS2 in my case) against the installed files, and it fails:
```
CMake Error at /usr/share/cmake-3.10/Modules/CMakeFindDependencyMacro.cmake:48 (find_package):
  By not providing "FindPCRE.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "PCRE", but
  CMake did not find one.

  Could not find a package configuration file provided by "PCRE" with any of
  the following names:

    PCREConfig.cmake
    pcre-config.cmake

  Add the installation prefix of "PCRE" to CMAKE_PREFIX_PATH or set
  "PCRE_DIR" to a directory containing one of the above files.  If "PCRE"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /root/cc_ws/sysroot_docker/usr/local/lib/cmake/Poco/PocoFoundationConfig.cmake:4 (find_dependency)
  /root/cc_ws/sysroot_docker/usr/local/lib/cmake/Poco/PocoConfig.cmake:29 (find_package)
  CMakeLists.txt:6 (find_package)
```

The `FindPCRE.cmake` is apparently not located on a standard cmake search path...

**Comment by Bjoe** [comment]
> The FindPCRE.cmake is apparently not located on a standard cmake search path...
>
Ah sh**, that's right. `install/path/lib/cmake/Poco/` is only set in `CMAKE_PREFIX_PATH` ... 
unfortunately I didn't found yet any proper solution, I "hacked" in `PocoFoundationConfig` to set `CMAKE_MODULE_PATH` .... I'm not so happy with it.
Please test ... sorry that I bother you again to test this, but I didn't have any test environment yet. I already take a short look into the ROS2 project but I didn't find in the document where I can quickly setup this.

Btw. the project looks very interesting for me :+1:

**Comment by lmayencourt** [comment]
I don't know either how to install this properly, that why I used the other hacky way with `PCRE_NAME` on my initial PR...

If you want to test with ROS2, you can find the instruction on how to setup an environment with docker on the [ROS2 tuto](https://index.ros.org/doc/ros2/Tutorials/Cross-compilation/) page. You will have to replace : 
`docker run --name arm_sysroot arm_ros2:latest`
with 
 `docker run -it --name arm_sysroot arm_ros2:latest`
This will allow you to compile (slowly) and install Poco on the target filesystem. I think you need also to remove the pre-installed version of Poco...

I will also test it on my side.

**Comment by Bjoe** [comment]
I try to get in the cross compiling of ROS2, but this is was a bigger task for me. 
I create a simple example and it work for me.
I fixed also the `XML` and `SQL` build with `EXPAT` and `SQLite3`.
If you don't have any concern then please merge my changes into your branch that we can discuss teh pull request with the other poco devs.
I will also create a pull request for the `poco-1.9.1` branch.

**Comment by lmayencourt** [comment]
> I try to get in the cross compiling of ROS2, but this is was a bigger task for me.

Yes sorry it was just a suggestion. For someone without knowledge of the project, it is a big overhead to setup. Especially if it is just to try compiling Poco...

> I create a simple example and it work for me.

Nice ! Same for me with my ROS2 setup !

I can merge your changes and update my PR on the Poco repo if required ! Thanks for your help on this !

**Pull request merged**